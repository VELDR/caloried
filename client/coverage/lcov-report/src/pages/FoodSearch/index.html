
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/pages/FoodSearch</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/pages/FoodSearch</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">57.44% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>54/94</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">41.17% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>14/34</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">53.84% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>14/26</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">57.83% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>48/83</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="60" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 60%"></div><div class="cover-empty" style="width: 40%"></div></div>
	</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="constants.js"><a href="constants.js.html">constants.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.jsx"><a href="index.jsx.html">index.jsx</a></td>
	<td data-value="64.1" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 64%"></div><div class="cover-empty" style="width: 36%"></div></div>
	</td>
	<td data-value="64.1" class="pct medium">64.1%</td>
	<td data-value="39" class="abs medium">25/39</td>
	<td data-value="55" class="pct medium">55%</td>
	<td data-value="20" class="abs medium">11/20</td>
	<td data-value="45.45" class="pct low">45.45%</td>
	<td data-value="11" class="abs low">5/11</td>
	<td data-value="65.78" class="pct medium">65.78%</td>
	<td data-value="38" class="abs medium">25/38</td>
	</tr>

<tr>
	<td class="file medium" data-value="reducer.js"><a href="reducer.js.html">reducer.js</a></td>
	<td data-value="54.54" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 54%"></div><div class="cover-empty" style="width: 46%"></div></div>
	</td>
	<td data-value="54.54" class="pct medium">54.54%</td>
	<td data-value="11" class="abs medium">6/11</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="54.54" class="pct medium">54.54%</td>
	<td data-value="11" class="abs medium">6/11</td>
	</tr>

<tr>
	<td class="file low" data-value="saga.js"><a href="saga.js.html">saga.js</a></td>
	<td data-value="10.52" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 10%"></div><div class="cover-empty" style="width: 90%"></div></div>
	</td>
	<td data-value="10.52" class="pct low">10.52%</td>
	<td data-value="19" class="abs low">2/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="10.52" class="pct low">10.52%</td>
	<td data-value="19" class="abs low">2/19</td>
	</tr>

<tr>
	<td class="file high" data-value="selectors.js"><a href="selectors.js.html">selectors.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-12-29T03:25:37.630Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    