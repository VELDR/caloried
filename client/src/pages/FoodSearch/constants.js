export const SET_FOODS = 'FoodSearch/SET_FOODS';
export const GET_FOODS = 'FoodSearch/GET_FOODS';
export const SET_CURRENT_PAGE = 'FoodSearch/SET_CURRENT_PAGE';
export const CREATE_CUSTOM_FOOD = 'FoodSearch/CREATE_CUSTOM_FOOD';
export const CLEAR_FOODS = 'FoodSearch/CLEAR_FOODS';
