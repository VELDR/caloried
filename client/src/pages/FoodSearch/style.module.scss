@import '@styles/base';

.page {
  @include flex-center-column;
  padding: 1rem 1rem 0;
  .container {
    max-width: 50rem;
    border: 1px solid var(--color-text-secondary);
    @include apply-at-md-size {
      width: 40rem;
    }
    .header {
      @include flex-column;
      padding: 0.5rem 1rem;
      border-bottom: 1px solid var(--color-text-secondary);
      &__title {
        font-size: $font-size-large;
        font-weight: 500;
      }
      .category {
        @include flex-items-center(1rem);

        .categoryWrapper {
          display: table-cell;
          position: relative;
        }
        &__item {
          cursor: pointer;
          &:after {
            content: '';
            background-color: var(--color-primary);
            width: 0;
            height: 2px;
            position: absolute;
            bottom: 0;
            left: 50%;
            transition:
              width 0.3s ease 0s,
              left 0.3s ease 0s;
          }
          &:hover {
            color: var(--color-primary);
          }
          &:hover:after {
            width: 100%;
            left: 0;
          }
        }
        .active {
          color: var(--color-primary);
          font-weight: 700;
          &:after {
            content: '';
            background-color: var(--color-primary);
            width: 100%;
            left: 0;
            height: 2px;
            position: absolute;
            bottom: 0;
            transition:
              width 0.3s ease 0s,
              left 0.3s ease 0s;
          }
        }
      }
    }
    .noResults {
      @include flex-center-column;
      padding: 2rem;
      text-align: center;
      font-size: $font-size-regular;
      @include apply-at-md-size {
        font-size: $font-size-medium;
      }
      img {
        @include box(75%);
        @include apply-at-md-size {
          @include box(20rem);
        }
      }
    }
  }
  .pagination {
    @include flex-center;
    padding: 1rem;
    ul {
      li {
        button {
          font-family: Poppins, 'sans-serif' !important;
          color: var(--color-text-primary);
          &:focus {
            color: var(--color-primary);
            border: 1px solid var(--color-primary);
            font-weight: 700;
          }
          &:hover {
            color: var(--color-primary);
            border: 1px solid var(--color-primary);
          }
        }
      }
    }
  }
}
