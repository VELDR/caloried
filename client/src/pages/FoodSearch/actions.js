import { CLEAR_FOODS, CREATE_CUSTOM_FOOD, GET_FOODS, SET_CURRENT_PAGE, SET_FOODS } from './constants';

export const getFoods = (query, page, pageSize, category, token) => ({
  type: GET_FOODS,
  query,
  page,
  pageSize,
  category,
  token,
});

export const setFoods = (foods, totalItems, token) => ({
  type: SET_FOODS,
  foods,
  totalItems,
  token,
});

export const setCurrentPage = (page) => ({
  type: SET_CURRENT_PAGE,
  page,
});

export const createCustomFood = (data, token, callback) => ({
  type: CREATE_CUSTOM_FOOD,
  data,
  token,
  callback,
});

export const clearFoods = () => ({
  type: CLEAR_FOODS,
});
