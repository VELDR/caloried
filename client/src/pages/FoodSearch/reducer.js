import { produce } from 'immer';
import { SET_CURRENT_PAGE, SET_FOODS } from './constants';

export const initialState = {
  foods: null,
  totalItems: 0,
  currentPage: 1,
  pageSize: 5,
};

export const storedKey = [];

const foodSearchReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_FOODS:
        draft.foods = action.foods;
        draft.totalItems = action.totalItems;
        break;
      case SET_CURRENT_PAGE:
        draft.currentPage = action.page;
        break;
    }
  });

export default foodSearchReducer;
