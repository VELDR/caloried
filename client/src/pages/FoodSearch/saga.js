import { setLoading, showPopup } from '@containers/App/actions';
import { getFoodsApi } from '@domain/api';
import toast from 'react-hot-toast';
import { takeLatest, call, put } from 'redux-saga/effects';
import { GET_FOODS } from './constants';
import { setFoods } from './actions';

function* doGetFoods({ query, page, pageSize, token, category }) {
  yield put(setLoading(true));
  try {
    const response = yield call(getFoodsApi, query, page, pageSize, token, category);
    yield put(setFoods(response.items, response.totalItems, token));
  } catch (error) {
    if (error.response && error.response.data) {
      toast.error(error.response.data.message);
    } else {
      yield put(showPopup());
    }
  } finally {
    yield put(setLoading(false));
  }
}

export default function* foodSearchSaga() {
  yield takeLatest(GET_FOODS, doGetFoods);
}
