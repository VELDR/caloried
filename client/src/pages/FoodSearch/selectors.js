import { createSelector } from 'reselect';
import { initialState } from '@containers/Client/reducer';

const selectFoodSearchState = (state) => state.foodSearch || initialState;

export const selectFoods = createSelector(selectFoodSearchState, (state) => state.foods);
export const selectCurrentPage = createSelector(selectFoodSearchState, (state) => state.currentPage);
export const selectPageSize = createSelector(selectFoodSearchState, (state) => state.pageSize);
export const selectTotalItems = createSelector(selectFoodSearchState, (state) => state.totalItems);
