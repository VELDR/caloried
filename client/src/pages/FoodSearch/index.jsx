import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';
import { connect, useDispatch } from 'react-redux';
import { useEffect, useState } from 'react';
import { useNavigate, useSearchParams } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import { Pagination } from '@mui/material';
import noResults from '@static/images/no-result.svg';
import noFood from '@static/images/no-food.svg';
import { isRoleMatch } from '@utils/authUtils';
import { Upload } from '@mui/icons-material';

import CustomFoodForm from '@components/CustomFoodForm';
import FoodCard from '@components/FoodCard';
import PrimaryButton from '@components/ui/PrimaryButton';

import { selectToken } from '@containers/Client/selectors';
import { selectCurrentPage, selectFoods, selectPageSize, selectTotalItems } from './selectors';
import { clearFoods, getFoods, setCurrentPage } from './actions';

import classes from './style.module.scss';

const FoodSearch = ({ token, foods, currentPage, pageSize, totalItems, intl: { formatMessage } }) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [selectedCategory, setSelectedCategory] = useState('All');
  const [searchParams, setSearchParams] = useSearchParams();
  const query = searchParams.get('query');
  const page = searchParams.get('page') || currentPage;
  const size = searchParams.get('pageSize') || pageSize;
  const totalPages = Math.ceil(totalItems / pageSize);

  const [isModalOpen, setIsModalOpen] = useState(false);

  const handleOpenClick = () => {
    setIsModalOpen(true);
  };

  const handleCloseModal = () => {
    setIsModalOpen(false);
  };

  useEffect(() => {
    if (query) {
      if (isRoleMatch(token, 'user')) {
        dispatch(getFoods(query, page, size, selectedCategory, token));
      }
    } else {
      dispatch(clearFoods());
    }
  }, [dispatch, query, page, size, selectedCategory, token]);

  const handlePageChange = (event, newPage) => {
    if (query) {
      setSearchParams({ query, page: newPage, pageSize: size });
      dispatch(setCurrentPage(newPage));
      dispatch(getFoods(query, newPage, size, selectedCategory, token));
    }
  };

  const handleCategoryChange = (category) => {
    setSelectedCategory(category);
    if (query) {
      setSearchParams({ query, page: 1, pageSize: size });
      dispatch(setCurrentPage(1));
      dispatch(getFoods(query, 1, size, category, token));
    }
  };

  const handleFoodCardClick = (food) => {
    navigate(`/food/${food?.type}/${encodeURIComponent(food?.foodName)}`);
  };

  return (
    <div className={classes.page} data-testid="food-search">
      <div className={classes.container}>
        <div className={classes.header}>
          <div className={classes.header__title}>
            {query
              ? `${formatMessage({ id: 'app_search_results' })} "${query}"`
              : formatMessage({ id: 'app_type_to_search' })}
          </div>
          <div className={classes.category}>
            {['All', 'Common', 'Branded', 'Custom'].map((category) => (
              <div className={classes.categoryWrapper} key={category}>
                <div
                  className={
                    selectedCategory === category
                      ? `${classes.active} ${classes.category__item}`
                      : classes.category__item
                  }
                  onClick={() => handleCategoryChange(category)}
                >
                  {category}
                </div>
              </div>
            ))}
          </div>
        </div>
        {foods && foods.length > 0 ? (
          foods.map((food, index) => <FoodCard key={index} food={food} onClick={() => handleFoodCardClick(food)} />)
        ) : (
          <div className={classes.noResults}>
            <img src={noResults} alt="" />
            <FormattedMessage id="app_blank_result" />
          </div>
        )}
      </div>
      <Pagination
        className={classes.pagination}
        count={totalPages}
        page={currentPage}
        onChange={handlePageChange}
        shape="rounded"
      />
      <div className={classes.addCustom}>
        <img src={noFood} alt="" />
        <div className={classes.addCustom__description}>
          <FormattedMessage id="app_no_food_description" />
        </div>
        <PrimaryButton isSubmit={false} className={classes.addCustom__button} onClick={handleOpenClick}>
          <Upload />
          <FormattedMessage id="app_submit_food" />
        </PrimaryButton>
      </div>
      <CustomFoodForm
        open={isModalOpen}
        onClose={handleCloseModal}
        token={token}
        query={query}
        page={page}
        size={size}
        selectedCategory={selectedCategory}
      />
    </div>
  );
};

FoodSearch.propTypes = {
  token: PropTypes.string,
  foods: PropTypes.array,
  currentPage: PropTypes.number,
  pageSize: PropTypes.number,
  totalItems: PropTypes.number,
  intl: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  token: selectToken,
  foods: selectFoods,
  currentPage: selectCurrentPage,
  pageSize: selectPageSize,
  totalItems: selectTotalItems,
});

export default injectIntl(connect(mapStateToProps)(FoodSearch));
