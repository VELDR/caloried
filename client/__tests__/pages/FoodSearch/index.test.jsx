import FoodSearch from '@pages/FoodSearch';
import { fireEvent, render } from '@utils/testHelper';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
  useSearchParams: () => [new URLSearchParams(), () => jest.fn()],
}));

let wrapper;

beforeEach(() => {
  wrapper = render(<FoodSearch />);
});

describe('FoodSearch Page', () => {
  it('should render the page correctly', () => {
    const { getByTestId, getByText } = wrapper;

    expect(getByTestId('food-search')).toBeInTheDocument();
    expect(getByText('All')).toBeInTheDocument();
    expect(getByText('Common')).toBeInTheDocument();
    expect(getByText('Branded')).toBeInTheDocument();
  });

  it('should change the category class to active when a category is clicked', () => {
    const { getByText } = wrapper;

    fireEvent.click(getByText('Common'));

    expect(getByText('Common')).toHaveClass('active');
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
