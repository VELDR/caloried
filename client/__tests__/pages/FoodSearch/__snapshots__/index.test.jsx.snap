// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodSearch Page should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page"
        data-testid="food-search"
      >
        <div
          class="container"
        >
          <div
            class="header"
          >
            <div
              class="header__title"
            >
              Ketik untuk mencari makanan
            </div>
            <div
              class="category"
            >
              <div
                class="categoryWrapper"
              >
                <div
                  class="active category__item"
                >
                  All
                </div>
              </div>
              <div
                class="categoryWrapper"
              >
                <div
                  class="category__item"
                >
                  Common
                </div>
              </div>
              <div
                class="categoryWrapper"
              >
                <div
                  class="category__item"
                >
                  Branded
                </div>
              </div>
              <div
                class="categoryWrapper"
              >
                <div
                  class="category__item"
                >
                  Custom
                </div>
              </div>
            </div>
          </div>
          <div
            class="noResults"
          >
            <img
              alt=""
            />
            Kami tidak menemukan apa-apa di sini. Mungkin coba kata kata yang lebih umum?
          </div>
        </div>
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root MuiPagination-text pagination css-1oj2twp-MuiPagination-root"
        >
          <ul
            class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-10w330c-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateBeforeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-10w330c-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateNextIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
        <div
          class="addCustom"
        >
          <img
            alt=""
          />
          <div
            class="addCustom__description"
          >
            Belum menemukan hidangan yang diinginkan?
          </div>
          <button
            class="button addCustom__button"
            data-testid="primary-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="UploadIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 20h14v-2H5v2zm0-10h4v6h6v-6h4l-7-7-7 7z"
              />
            </svg>
            Tambah makanan
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page"
      data-testid="food-search"
    >
      <div
        class="container"
      >
        <div
          class="header"
        >
          <div
            class="header__title"
          >
            Ketik untuk mencari makanan
          </div>
          <div
            class="category"
          >
            <div
              class="categoryWrapper"
            >
              <div
                class="active category__item"
              >
                All
              </div>
            </div>
            <div
              class="categoryWrapper"
            >
              <div
                class="category__item"
              >
                Common
              </div>
            </div>
            <div
              class="categoryWrapper"
            >
              <div
                class="category__item"
              >
                Branded
              </div>
            </div>
            <div
              class="categoryWrapper"
            >
              <div
                class="category__item"
              >
                Custom
              </div>
            </div>
          </div>
        </div>
        <div
          class="noResults"
        >
          <img
            alt=""
          />
          Kami tidak menemukan apa-apa di sini. Mungkin coba kata kata yang lebih umum?
        </div>
      </div>
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-text pagination css-1oj2twp-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-10w330c-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-10w330c-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
      <div
        class="addCustom"
      >
        <img
          alt=""
        />
        <div
          class="addCustom__description"
        >
          Belum menemukan hidangan yang diinginkan?
        </div>
        <button
          class="button addCustom__button"
          data-testid="primary-button"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="UploadIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 20h14v-2H5v2zm0-10h4v6h6v-6h4l-7-7-7 7z"
            />
          </svg>
          Tambah makanan
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
